gemrb/core/GameScript/GSUtils.cpp

```diff
@@ -2079,164 +2079,100 @@ Trigger *GenerateTriggerCore(const char *src, const char *str, int trIndex, int
 	return newTrigger;
 }
 
-void SetVariable(Scriptable* Sender, const char* VarName, const char* Context, ieDword value)
+void SetVariable(Scriptable* Sender, const char* VarName, ieDword value, const char* Context)
 {
 	char newVarName[8+33];
-
-	ScriptDebugLog(ID_VARIABLES, "Setting variable(\"%s%s\", %d)", Context, VarName, value);
-
-	strlcpy( newVarName, Context, 7 );
-	if (strnicmp( newVarName, "MYAREA", 6 ) == 0) {
-		Sender->GetCurrentArea()->locals->SetAt( VarName, value, NoCreate );
-		return;
-	}
-	if (strnicmp( newVarName, "LOCALS", 6 ) == 0) {
-		Sender->locals->SetAt( VarName, value, NoCreate );
-		return;
-	}
-	Game *game = core->GetGame();
-	if (HasKaputz && !strnicmp(newVarName,"KAPUTZ",6) ) {
-		game->kaputz->SetAt( VarName, value );
-		return;
-	}
-
-	if (strnicmp(newVarName,"GLOBAL",6) ) {
-		Map *map=game->GetMap(game->FindMap(newVarName));
-		if (map) {
-			map->locals->SetAt( VarName, value, NoCreate);
-		} else if (core->InDebugMode(ID_VARIABLES)) {
-			Log(WARNING, "GameScript", "Invalid variable %s %s in setvariable",
-				Context, VarName);
-		}
+	
+	const char *poi = VarName;
+	if (Context == nullptr) {
+		poi = &VarName[6];
+		//some HoW triggers use a : to separate the scope from the variable name
+		if (*poi==':') {
+			poi++;
+		}
+		strlcpy(newVarName, VarName, 7);
 	} else {
-		game->locals->SetAt(VarName, value, NoCreate);
-	}
-}
-
-void SetVariable(Scriptable* Sender, const char* VarName, ieDword value)
-{
-	char newVarName[8];
-	const char *poi;
-
-	poi = &VarName[6];
-	//some HoW triggers use a : to separate the scope from the variable name
-	if (*poi==':') {
-		poi++;
+		strlcpy(newVarName, Context, 7);
 	}
-
-	ScriptDebugLog(ID_VARIABLES, "Setting variable(\"%s\", %d)", VarName, value);
-
-	strlcpy( newVarName, VarName, 7 );
-	if (stricmp( newVarName, "MYAREA" ) == 0) {
+	ScriptDebugLog(ID_VARIABLES, "Setting variable(\"%s%s\", %d)", Context, VarName, value);
+	
+	if (strnicmp( newVarName, "MYAREA", 6) == 0) {
 		Sender->GetCurrentArea()->locals->SetAt( poi, value, NoCreate );
 		return;
 	}
-	if (stricmp( newVarName, "LOCALS" ) == 0) {
+	if (strnicmp( newVarName, "LOCALS", 6) == 0) {
 		Sender->locals->SetAt( poi, value, NoCreate );
 		return;
 	}
 	Game *game = core->GetGame();
-	if (HasKaputz && !stricmp(newVarName,"KAPUTZ") ) {
+	if (HasKaputz && !strnicmp(newVarName, "KAPUTZ", 6)) {
+		// FIXME?: NoCreate used to ony be passed when Context was null
+		// It was changed during consolidation because I couldnt think of why that difference was purposeful
+		// it felt like a bug, but thats only a hunch
```

Comment:
It's fine, you can remove the fixme. PST doesn't want any new variables and if we've been lucky not to hit it here, it doesn't matter — let's be consistent.